### ISSUE TITLE
Calling `Job.promise` throws error if job exceeds pool limit

### ISSUE THREAD

**Issue by daiheitan** [andywer/threads.js#12]
Currently when `job.send` is called, it is emitting `readyToRun` to wait for a thread, so when calling `.promise()` on it, as the job has no `this.thread`, it will throw `Cannot return promise, since job is not executed.`

`.promise` should wait until the thread is assigned to the job.

**Comment by maysale01** [comment]
+1

**Comment by andywer** [comment]
@daiheitan  @maysale01 Could you please try the `develop` branch? Should be fixed now.

**Comment by maysale01** [comment]
@andywer This issue appears to be resolved, however, I had to change my implementation because each message is being consumed by all available threads or the thread state is being overwritten in the midst of an asynchronous action (not sure). This is probably an issue with my implementation, see an example below:

```js
function testPools () {
    const pool = new Threads.Pool;
    const poolPromises = [];

    // Worker time
    pool.run(testHandler);

    for (let i = 0; i < 10; i++) {
        poolPromises.push(
            pool.send(`Test - ${i}`).promise()
        );
    }

    Ember.RSVP.all(poolPromises)
    .then((response) => {
        this.debug(response);
    })
}

function testHandler(input, done) {
    console.log(input);
    done(input);
}
```

<img width="1430" alt="screen shot 2016-01-27 at 11 28 02" src="https://cloud.githubusercontent.com/assets/3009203/12620248/8e8fc306-c4e9-11e5-82e6-ff4b26af647c.png">

**Comment by andywer** [comment]
@maysale01 Hey ho! I just merged your PR and released 0.6.0. Is this issue fixed now?
